projects/storefrontlib/src/layout/direction/direction.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  ConfigInitializerService,
+  LanguageService,
+  WindowRef,
+} from '@spartacus/core';
+import { Subscription } from 'rxjs';
+import { DirectionMode, LayoutDirection } from '../config/direction.model';
+import { LayoutConfig } from '../config/layout-config';
+
+/**
+ * The `DirectionService` can be used to add the direction to the overall storefront or individual elements.
+ * By default, the direction is added to the `html` element (i.e. `<html dir="ltr">`). The API of this service
+ * does however provide methods to add direction to individual elements if needed.
+ *
+ * The direction is configurable in the `LayoutDirection` configuration. The `detect` mode is recommended for
+ * a (multi) language driven setup.
+ *
+ * To react to the active language, the service subscribes to the active language in the initialize method. This
+ * is called from an APP_INITIALIZER method and should only happen once.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DirectionService implements OnDestroy {
+  protected config: LayoutDirection;
+  protected startsDetecting = false;
+
+  protected subscription = new Subscription();
+
+  constructor(
+    protected configInit: ConfigInitializerService,
+    protected languageService: LanguageService,
+    protected winRef: WindowRef
+  ) {}
+
+  /**
+   * Initializes the layout direction for the storefront.
+   */
+  initialize(): void {
```

Comment:
I've just realied it's a breaking change in the DOM. This method manipulates the `dir` attribute of the `html` the regardless customer already has a custom dynamic mechanism (or just hardcoded value RTL) for this attribute. This customer may like to have RTL, while not having active langauge `ar` or `he`.

So IMO it should be protected with a feature level `2.1`.

Reply:
Agreed. I'll drop the default values while we're not on 3.x.